generate-default-operators/src/GenerateDefaultOperatorsModule.purs

```diff
@@ -0,0 +1,149 @@
+module GenerateDefaultOperatorsModule where
+
+import Prelude
+
+import Data.Argonaut.Core (Json)
+import Data.Argonaut.Decode as Data.Argonaut.Decode
+import Data.Array (foldMap, mapWithIndex)
+import Data.Array as Array
+import Data.Either (Either(..), either)
+import Data.Maybe (Maybe(..))
+import Data.Number (infinity)
+import Data.String (Pattern(..))
+import Data.String as Str
+import Data.String as String
+import Data.String.Regex as Regex
+import Data.String.Regex.Flags (noFlags)
+import Data.String.Regex.Unsafe (unsafeRegex)
+import Effect (Effect)
+import Effect.Aff (Aff, error, runAff_, throwError)
+import Effect.Class (liftEffect)
+import Effect.Class.Console as Console
+import Effect.Exception (throwException)
+import Foreign.Object (Object)
+import Foreign.Object as Foreign.Object
+import Node.Buffer as Buffer
+import Node.ChildProcess as ChildProcess
+import Node.ChildProcess as Exec
+import Node.ChildProcess.Types as Node.ChildProcess.Types
+import Node.Encoding (Encoding(..))
+import Node.FS.Aff (readdir, stat, writeTextFile)
+import Node.FS.Stats as FS
+import Node.Library.Execa as Node.Library.Execa
+import Node.Path (FilePath)
+import Node.Path as Path
+import Node.Process as Node.Process
+
+foreign import tmpdir :: String -> Effect String
+
+main :: Effect Unit
+main = runAff_ (either throwException mempty) do
+  cwdPath <- liftEffect Node.Process.cwd
+  tmpPath <- liftEffect $ tmpdir "purs-tidy-generate-default-operators-"
+  liftEffect $ Console.log $ "Working in " <> tmpPath
+
+  writeTextFile UTF8 (Path.concat [ tmpPath, "spago.yaml" ]) defaultSpagoYaml
+  writeTextFile UTF8 (Path.concat [ tmpPath, "package.json" ]) defaultPackageJson
+  liftEffect $ Console.log $ "running 'npm install'"
+  _ <- liftEffect $ ChildProcess.execSync' "npm install" (_ { cwd = Just tmpPath })
+
+  liftEffect $ Console.log $ "running 'spago ls packages --json'"
+  s <- liftEffect $ Buffer.toString UTF8 =<< Exec.execSync' "spago ls packages --json" (_ { cwd = Just tmpPath })
+
+  packages <- case Data.Argonaut.Decode.decodeJson =<< Data.Argonaut.Decode.parseJson s of
+    Left err -> throwError $ error $ Data.Argonaut.Decode.printJsonDecodeError err
+    Right (object :: Object Json) -> pure $ Foreign.Object.keys object
+
+  liftEffect $ Console.log $ "running 'spago install " <> Str.joinWith " " packages <> "'"
+  _ <- liftEffect $ Exec.execSync' ("spago install " <> Str.joinWith " " packages) (_ { cwd = Just tmpPath })
+  pursFiles <- getPursFiles 0 (tmpPath <> "/.spago")
```

Comment:
Does spago no longer provide a way to get the globs of library sources?

Reply:
done